src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -181,7 +181,7 @@ protected override void ApplyResponseGrant()
                     if (signInContext.Properties.IsPersistent)
                     {
                         var expiresUtc = signInContext.Properties.ExpiresUtc ?? issuedUtc.Add(Options.ExpireTimeSpan);
-                        signInContext.CookieOptions.Expires = expiresUtc.ToUniversalTime().DateTime;
+                        signInContext.CookieOptions.Expires = expiresUtc.LocalDateTime;
```

Comment:
Because the `Kind` property of `expiresUtc.ToUniversalTime().DateTime` is `DateTimeKind.Unspecified` rather than `DateTimeKind.Local`. So there is a problem in this code:
```c#
// ChunkingCookieManager.cs, line 129
var template = new SetCookieHeaderValue(escapedKey)
{
    Domain = options.Domain,

    // convert DateTime to DateTimeOffset implicitly here
    // because options.Expires.Kind == DateTimeKind.Unspecified,
    // options.Expires is treated as a local DateTime during the conversion while it's already a UTC DateTime
    Expires = options.Expires,

    HttpOnly = options.HttpOnly,
    Path = options.Path,
    Secure = options.Secure,
};
```